src/SearchHistoryModalBody.tsx

```diff
@@ -0,0 +1,106 @@
+import styles from './SearchHistoryModal.module.css';
+import { useRef } from 'react';
+import { useIsMobileLayout } from './theme/useIsMobileLayout';
+import ButtonIcon from './ButtonIcon';
+import { X } from 'lucide-react';
+import { GroupedSearchHistories } from './types';
+import { formatContentSnippet, formatConversationTitle } from './utils/helpers';
+
+interface SearchModalBodyProps {
+  groupedConversations: GroupedSearchHistories;
+  onSelectConversation: (id: string) => void;
+  handleSearchTermChange: (searchTerm: string) => void;
+  inputValue: string;
+  onClose: () => void;
+}
+
+export const SearchHistoryModalBody = ({
+  groupedConversations,
+  onSelectConversation,
+  handleSearchTermChange,
+  inputValue,
+  onClose,
+}: SearchModalBodyProps) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      onClose();
+      return;
+    }
+
+    if (e.key === 'Enter') {
+      //  if there's only one remaining search match, allow it to be selected with the enter key
+      const allConversations = Object.values(groupedConversations).flat();
+      if (allConversations.length === 1) {
+        onSelectConversation(allConversations[0].id);
+        onClose();
+      }
+    }
```

Comment:
This is new, useful functionality - allowing the user to click enter with only a single remaining history. I added unit tests for it below.